demo-app/src/main/java/cash/z/ecc/android/sdk/demoapp/ui/screen/home/viewmodel/WalletViewModel.kt

```diff
@@ -246,6 +246,31 @@ class WalletViewModel(application: Application) : AndroidViewModel(application)
         }
     }
 
+    /**
+     * Synchronously provides proposal object for the given [spendingKey] and [uri] objects
+     */
+    fun getSendProposalFromUri(uri: String): Proposal? {
+        if (sendState.value is SendState.Sending) {
+            return null
+        }
+
+        val synchronizer = synchronizer.value
+
+        return if (null != synchronizer) {
+            // Calling the proposal API within a blocking coroutine should be fine for the showcase purpose
+            runBlocking {
+                val spendingKey = spendingKey.filterNotNull().first()
+                kotlin.runCatching {
+                    synchronizer.proposeFulfillingPaymentUri(spendingKey.account, uri)
```

Comment:
I'm a little concerned that this doesn't provide any mechanism to report back to the caller what is actually wrong with the proposal - in particular, it seems pretty important to be able to distinguish an "Insufficient funds" error from an invalid proposal URI.

Reply:
Correct. This junk of code is meant to be used only to quickly check that the get-proposal method returns a result in the simplest way, and thus, we do not display potential errors here. Note that this is just a Demo app part. It does not affect how we handle the Proposal object in SDK and Zashi. Many of the Demo app parts do not align with best practices, for which we have Zashi as our showcase.